### PR TITLE
fix and test linear decoder notebook

### LINKED ISSUE #474: 'GeneExpressionDataset' has no attribute 'get_attributes_from_matrix'
I am trying to run through your notebook "https://github.com/YosefLab/scVI/blob/master/tests/notebooks/linear_decoder.ipynb", and am encountering the error "'GeneExpressionDataset' has no attribute 'get_attributes_from_matrix'" when trying to run the step "X, local_means, local_vars, batch_indices, labels = GeneExpressionDataset.get_attributes_from_matrix(adata.X)"

Has this function been deprecated and removed in favour of another one or am I doing something wrong here?

I am using a virtual environment with Anaconda, and cloned and installed this project using the setup.py included in the repo.

### PR DESCRIPTION
**Pull request by adamgayoso**
- Fixes #474

### CODE REVIEW

**Comment by adamgayoso** [comment]
I believe there is other old code in here that needs to be fixed and then the entire notebook rerun

**Comment by romain-lopez** [comment]
Thanks for this Adam! To follow up on this, I am looking into the CI report but there is nothing helpful in there.

File: tests/notebooks/linear_decoder.ipynb [hunk 1]
```diff
@@ -303,6 +246,7 @@
     }
    ],
    "source": [
+    "n_epochs = 50 if n_epochs_all is None else n_epochs_all\n",
     "trainer.train(n_epochs=50)"
```
Comment (romain-lopez) [inline tests/notebooks/linear_decoder.ipynb]: should this be n_epochs=n_epochs ?

**Comment by romain-lopez** [comment]
Please look at my one comment about n_epochs, this can be then merged

File: tests/notebooks/linear_decoder.ipynb (hunk 1 above)
Comment (adamgayoso) [reply tests/notebooks/linear_decoder.ipynb]: It should't be, this allows n_epochs to be set to 1 in test mode. Our testing code finds n_epochs_all and makes it 1.

**Pull request merged**